File: android_app/lib/services/api_constants.dart
const String baseUrl = 'https://jusdown.onrender.com/api';

class ApiConstants {
  // Auth endpoints
  static const String authYoutube = '/auth/youtube';
  static const String authCallback = '/auth/callback';
  static const String authRefresh = '/auth/refresh';
  static const String authChannels = '/auth/channels';
  static const String authSignout = '/auth/signout';

  // Scheduler endpoints
  static const String schedulers = '/schedulers';
  static String schedulerById(String id) => '/schedulers/$id';
  static String schedulerToggle(String id) => '/schedulers/$id/toggle';

  // Video endpoints
  static const String videos = '/videos';
  static String videoById(String id) => '/videos/$id';
  static String videoThumbnail(String id) => '/videos/$id/thumbnail';
  static String videoUploadNow(String id) => '/videos/$id/upload-now';

  // Settings endpoints
  static const String settings = '/settings';
  static const String settingsSunoKey = '/settings/suno-key';
  static const String settingsQuota = '/settings/quota';
  static const String settingsUpgrade = '/settings/upgrade';

  // Webhooks
  static const String webhookVideoDeleted = '/webhooks/video-deleted';
}

File: android_app/app/src/main/java/com/autosong/studio/data/remote/ApiConstants.kt
package com.autosong.studio.data.remote

object ApiConstants {
    const val BASE_URL = "https://51.222.156.30.sslip.io/api/"

    // Auth endpoints
    const val AUTH_YOUTUBE = "auth/youtube"
    const val AUTH_CALLBACK = "auth/callback"
    const val AUTH_REFRESH = "auth/refresh"
    const val AUTH_CHANNELS = "auth/channels"
    const val AUTH_SIGNOUT = "auth/signout"

    // Scheduler endpoints
    const val SCHEDULERS = "schedulers"
    fun schedulerById(id: String) = "schedulers/$id"
    fun schedulerToggle(id: String) = "schedulers/$id/toggle"

    // Video endpoints
    const val VIDEOS = "videos"
    fun videoById(id: String) = "videos/$id"
    fun videoThumbnail(id: String) = "videos/$id/thumbnail"
    fun videoUploadNow(id: String) = "videos/$id/upload-now"
    fun videoStream(id: String) = "videos/$id/stream"

    // Settings endpoints
    const val SETTINGS = "settings"
    const val SETTINGS_SUNO_KEY = "settings/suno-key"
    const val SETTINGS_QUOTA = "settings/quota"
    const val SETTINGS_UPGRADE = "settings/upgrade"

    // Webhooks
    const val WEBHOOK_VIDEO_DELETED = "webhooks/video-deleted"
}
